graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,53 +1,174 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from collections import namedtuple
+from dataclasses import dataclass, field
+from typing import List, NamedTuple, Tuple
+
+from graphql import DocumentNode
 
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
-from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
+from ..cost_estimation.helpers import is_uuid4_type
+from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
 from ..exceptions import GraphQLError
+from ..schema.schema_info import QueryPlanningSchemaInfo
+
+
+@dataclass
+class PaginationAdvisory(object):
+    message: str
```

Comment:
If you want to set this field in a `__post_init__` then consider making it an abstract base class and set `message: str = field(init=False)` here.